rbac/management/role/definer.py

```diff
@@ -29,37 +30,30 @@
 
 def _make_role(tenant, data, update=False):
     """Create the role object in the database."""
-    with tenant_context(tenant):
-        name = data.pop("name")
-        description = data.pop("description", None)
-        access_list = data.pop("access")
-        version = data.pop("version", 1)
-        version_diff = False
-        is_platform_default = data.pop("platform_default", False)
-        if update:
-            role, created = Role.objects.filter(name=name).get_or_create(name=name)
-            version_diff = version != role.version
-            if created or (not created and version_diff):
-                logger.info("Updating role %s for tenant %s.", name, tenant.schema_name)
-                role.description = description
-                role.system = True
-                role.version = version
-                role.platform_default = is_platform_default
-                role.save()
-                role.access.all().delete()
-        else:
-            role = Role.objects.create(
-                name=name, description=description, system=True, version=version, platform_default=is_platform_default
-            )
-            logger.info("Creating role %s for tenant %s.", name, tenant.schema_name)
-        if not update or (update and version_diff):
-            for access_item in access_list:
-                resource_def_list = access_item.pop("resourceDefinitions", [])
-                access_obj = Access.objects.create(**access_item, role=role)
-                access_obj.save()
-                for resource_def_item in resource_def_list:
-                    res_def = ResourceDefinition.objects.create(**resource_def_item, access=access_obj)
-                    res_def.save()
+    name = data.pop("name")
+    version_diff = False
+    access_list = data.get("access")
+    defaults = dict(
+        description=data.get("description", None),
+        system=True,
+        version=data.get("version", 1),
+        platform_default=data.get("platform_default", False),
+    )
+    logger.info("%s role %s for tenant %s.", "Updating" if update else "Creating", name, tenant.schema_name)
+    role, created = Role.objects.get_or_create(name=name, defaults=defaults)
+    version_diff = defaults["version"] != role.version
+    if created or (not created and version_diff):
+        logger.info("Updating role %s for tenant %s.", name, tenant.schema_name)
```

Comment:
Doh! Sorry, I had a review in progress and just noticed the comments from @astrozzc and your latest commit. 😄 Disregard.

Reply:
Though, this does still show the "Updating" message on the noops.

Reply:
What would you like the log messages to communicate?

Reply:
I think it's probably fine as is, unless we wanted to avoid the logging noise when there is no update/create, and only log the updating/creating message in each of those conditions.